var/spack/repos/builtin/packages/npb/package.py

```diff
@@ -66,20 +76,39 @@ class Npb(MakefilePackage):
     )
 
     # TODO: Combine these into a single mutually exclusive variant
-    variant('mpi',    default=True,  description='Build the MPI implementation')
-    variant('openmp', default=False, description='Build the OpenMP implementation')
-    variant('serial', default=False, description='Build the serial version')
-
-    # TODO: Convert these to non-exclusive multi-valued variants
-    variant('names', default=','.join(valid_names),
-            description='Benchmark names (comma separated list)')
-    variant('classes', default=','.join(valid_classes),
-            description='Benchmark classes (comma separated list)')
+    variant(
+        'implementation',
+        default='mpi',
+        values=('serial', 'mpi', 'openmp'),
+        description='Selects one among the available implementations'
+    )
+
+    variant(
+        'names',
+        default=','.join(valid_names),
+        values=valid_names,
```

Comment:
How does this work?  `valid_names` is a list of the possible names, but the value of the variant is going to be a string built by joining some subset of `valid_names` with commas.  Is the magic sauce [this bit here in `variant.py`](https://github.com/LLNL/spack/blob/develop/lib/spack/spack/variant.py#L221), which looks like it converts the comma-separated string back into a uniq'ed list?